robottelo/host_helpers/capsule_mixins.py

```diff
@@ -66,21 +71,57 @@ def wait_for_sync(self, timeout=600, start_time=None):
         # Assert that a task to sync lifecycle environment to the capsule
         # is started (or finished already)
         if start_time is None:
-            start_time = datetime.utcnow().replace(microsecond=0)
+            start_time = datetime.utcnow()
+        # 1s margin of safety for rounding
+        start_time = (
+            (start_time - timedelta(seconds=1))
+            .replace(microsecond=0)
+            .strftime('%Y-%m-%d %H:%M:%S UTC')
+        )
         logger.info(f"Waiting for capsule {self.hostname} sync to finish ...")
         sync_status = self.nailgun_capsule.content_get_sync()
```

Comment:
Hmm, I've seen `'active_sync_tasks'` returned (and task being in progreess) so many times that I doubt it.

But looking into the nailgun entities it seems you are right - `_handle_response` really does some polling. Although, I guess we need to provide the `timeout` in the call to make it work, since default is `None`.

Reply:
Right, do we think we should keep this method @vsedmik @lpramuk ? 
Or just rely on `nailgun_capsule.content_get_sync()` and task assertions in every test using it here?

Reply:
@damoore044 Let's try how `nailgun_capsule.content_get_sync(synchronous=True, timeout=<something>)` works first and if it covers our needs just use it. If it does not, go with `wait_for_sync`.

Reply:
@vsedmik Will do for next commit :)

Reply:
Latest changes remove references of and modifications to `capsule_mixins::wait_for_sync()`.
Now there is a helper method in this module, `_capsule_sync_wait_assert()`;
- Where we assert some active sync tasks or a recently completed sync, and all fields updated properly.
- Most of the api tests use this method in place of older `wait_for_sync()`, after manually starting capsule sync task, or invoking with some parent task (cv promotion, repo sync etc).

Marking resolved